conda_libmamba_solver/index.py

```diff
@@ -282,7 +364,31 @@ def _load_channels(
             encoded_urls_to_channel[url] = channel
         urls_to_channel = encoded_urls_to_channel
 
-        urls_to_json_path_and_state = self._fetch_repodata_jsons(tuple(urls_to_channel.keys()))
+        if self.in_state and _is_sharded_repodata_enabled():
+            # try to make a subset of possible dependencies
+            # conda probably already has a suitable temporary directory?
+            # keep a reference so GC doesn't delete the directory
+            self.tmp_dir = tempfile.TemporaryDirectory("conda-shards")
+            self.tmp_path = Path(self.tmp_dir.name)
+            root_packages = (*self.in_state.installed.keys(), *self.in_state.requested)
+            channel_data = build_repodata_subset(root_packages, urls_to_channel)
+            begin = time.monotonic_ns()
+            # XXX use this instead of urls_to_json_path_and_state; append directly to channel_repo_infos?
+            _subset_infos = self._load_repo_info_from_repodata_dict(channel_data)
```

Comment:
If we are totally replacing `for url_w_cred, (json_path, state) in urls_to_json_path_and_state.items():` in shards mode, we could create the entire channel_repo_infos list here.
We probably preserve the channel order because of Python's ordered dict, to not worry about channel order.
We might want to add a sorted(package name keys) in _load_repo_info_from_repodata_dict "just in case"